### PR TITLE
feat: [CDS-44142]: add doNotDownsizeOldService field to EcsBlueGreenSwapTargetGroups step

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Add doNotDownsizeOldService field under spec field in EcsBlueGreenSwapTargetGroups step
- Pass environmentRef and infraDefinitionRef to clusters API to make it work in Runtime view when Service / Env V2 FF is true
- Do not allow Runtime / Expression type on checkbox fields that are present in EcsRollingDeploy step
- Add / Fix test cases

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 24 Issues 
* Code issue 24 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11587)

File: src/modules/75-cd/components/PipelineSteps/ECSBlueGreenCreateServiceStep/__tests__/ECSBlueGreenCreateServiceStepEdit.test.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+    const identifierEditIcon = queryByAttribute('data-icon', container, 'Edit')
+    expect(identifierEditIcon).toBeInTheDocument()
+
+    const nameInput = queryByNameAttribute('name') as HTMLInputElement
+    expect(nameInput).toBeInTheDocument()
+    expect(nameInput.value).toBe('')
+    act((): void => {
+      fireEvent.change(nameInput, { target: { value: 'Test Name' } })
+    })
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/ECSBlueGreenCreateServiceStep/__tests__/ECSBlueGreenCreateServiceStepEdit.test.tsx:138]: `fireEvent` is also wrapped in act 
https://github.com/testing-library/react-testing-library/blob/main/src/pure.js#L31

**Review by rameshx — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/ECSBlueGreenCreateServiceStep/__tests__/ECSBlueGreenCreateServiceStepEdit.test.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/ECSBlueGreenCreateServiceStep/__tests__/ECSBlueGreenCreateServiceStepEdit.test.tsx:138]: I will remove it in next PRs.

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 2]
```diff
@@ -1087,10 +1087,12 @@ export function StageInputSetFormInternal({
                           path={`${path}.environment.infrastructureDefinitions.${index}.inputs.spec`}
                           readonly={readonly}
                           stepViewType={viewType}
-                          customStepProps={getCustomStepProps(
-                            (deploymentStage?.deploymentType as StepType) || '',
-                            getString
-                          )}
+                          customStepProps={{
+                            ...getCustomStepProps((deploymentStage?.deploymentType as StepType) || '', getString),
+                            serviceRef: deploymentStage.service?.serviceRef,
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:1092]: `deploymentStage` can be undefined, use optional chaining

File: src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpec.tsx [hunk 3]
```diff
@@ -179,22 +182,22 @@ export class ECSInfraSpec extends PipelineStep<EcsInfrastructureStep> {
...
         />
       )
     } else if (stepViewType === StepViewType.InputVariable) {
       return (
         <VariablesListTable
-          data={(customStepProps as ECSInfraSpecVariableProps)?.variablesData?.infrastructureDefinition?.spec}
+          data={(customStepProps as ECSInfraSpecCustomStepProps)?.variablesData?.infrastructureDefinition?.spec}
           originalData={initialValues.infrastructureDefinition?.spec || initialValues}
-          metadataMap={(customStepProps as ECSInfraSpecVariableProps).metadataMap}
+          metadataMap={(customStepProps as ECSInfraSpecCustomStepProps).metadataMap}
```
Comment (depanshu-dhiman) [inline src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpec.tsx:200]: `customStepProps` can be `undefined`

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**